paimon-common/src/main/java/org/apache/paimon/io/cache/CachedRandomInputView.java

```diff
@@ -55,7 +55,7 @@ public CachedRandomInputView(File file, CacheManager cacheManager)
         this.file = new RandomAccessFile(file, "r");
         this.fileLength = file.length();
         this.cacheManager = cacheManager;
-        this.segments = new HashMap<>();
+        this.segments = new ConcurrentHashMap<>();
```

Comment:
What's this change for? Maybe you should create an another issue and add test case for this change

Reply:
@FangYongs  This is to fix ConcurrentModificationException threw by JDK11. 

Generally speaking, it's part of JDK11 supported work.

> What's this change for? Maybe you should create an another issue and add test case for this change

Reply:
@legendtkl, adds the test case to test this fix for `CachedRandomInputView` in JDK11?

Reply:
@SteNicholas  ok

Reply:
@SteNicholas  yes, once we replace computeIfAbsent log, it's ok to use HashMap. 

Let's keep the hashmap left.